iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -731,7 +745,6 @@ public IotHubClientProtocol getProtocol()
     public void sendMessages()
     {
         checkForExpiredMessages();
-        new Thread(() -> checkForOldMessages()).start();
```

Comment:
This method was called very frequently, so we were spawning a lot of threads. Now we just maintain one thread at a time and that thread sleeps for a while after cleaning up stale callbacks